### PR TITLE
17.0

### PR DESCRIPTION
**Pull request by 3mar-sabouni**

### CODE REVIEW

File: __manifest__.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+{
+    'name': "flower shop",
+
+    'summary': "",
```
Comment (mds-odoo) [inline __manifest__.py]: don't leave the summary unfilled

File: __manifest__.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        'views/template_product_inhert.xml',
+        'views/stock_production_lot_inhert.xml',
+        'views/warehouse_weather_views.xml',
+        'views/stock_warehouse_views.xml',
+        'reports/flower_sale_order_views.xml',
+        'data/ir_cron.xml',
+        'data/ir_actions_server.xml',
+        'data/ir_config_parameter_data.xml'
+        ],
+    'demo': [
```
Comment (mds-odoo) [inline __manifest__.py]: remove the whole thing if there is no  use for it

File: demo/demo.xml
Comment (mds-odoo) [inline demo/demo.xml:1]: remove the file if there is no use for it

File: views/product_product_views.xml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    <record id="sale_order_line_inhert_flower" model="ir.ui.view">
+        <field name="name">sale.order.line.inhert.flower</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='order_line']/tree/field[@name='product_id']" position="attributes">
+                <attribute name="domain">[('sale_ok', '=', True), '|', ('company_id', '=', False), ('company_id', '=', parent.company_id), ('is_flower', '=', True)]</attribute>
+            </xpath>
+        </field>
```
Comment (mds-odoo) [inline views/product_product_views.xml:48]: the domain is not working, it needs to be overridden from the model py file by inheriting the field and changing the domain

File: security/ir.model.access.csv [hunk 4]
```diff
@@ -0,0 +1,5 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_flower_shop_flower_flower,flower_shop.flower_shop,model_flower_flower,base.group_user,1,1,1,1
+access_flower_shop_flower_water,flower_shop.flower_water,model_flower_water,base.group_user,1,0,0,0
```
Comment (mds-odoo) [inline security/ir.model.access.csv:3]: you are having same access rights to the same groups, these are conflicting and are not allowing me to write

File: models/stock_lot.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    is_flower = fields.Boolean(related='product_id.is_flower', readonly=True)
+    water_ids = fields.One2many("flower.water", "serial_id")
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            product = self.env["product.product"].browse(vals["product_id"])
+            if product.sequence_id:
+                vals["name"] = product.sequence_id.next_by_id()
+        return super().create(vals_list)
```
Comment (mds-odoo) [inline models/stock_lot.py]: this needs to check if the product is a flower

File: models/stock_warehouse.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+                # Get forecast data between 9 AM to 6 PM for the current day
+                current_day_forecast = []
+                for entry in forecast_data['list']:
+                    dt = entry['dt']
+                    dt_time = datetime.fromtimestamp(dt)
+                    if dt_time.hour >= 9 and dt_time.hour <= 18:
+                        current_day_forecast.append(entry)
+
+                # Check rain volume and water flowers if necessary
+                watering_needed = False
```
Comment (mds-odoo) [inline models/stock_warehouse.py]: i would rather change the name to is_watered as it shows the meaning better

File: __manifest__.py (hunk 1 above)
Comment (3mar-sabouni) [reply __manifest__.py]: Noted sir..

File: views/product_product_views.xml (hunk 3 above)
Comment (3mar-sabouni) [reply views/product_product_views.xml:48]: the domain are working properly you just need to unfold the  field "product_id".

File: security/ir.model.access.csv (hunk 4 above)
Comment (3mar-sabouni) [reply security/ir.model.access.csv:3]: they are different models

File: models/stock_lot.py (hunk 5 above)
Comment (3mar-sabouni) [reply models/stock_lot.py]: Done..

File: models/stock_warehouse.py (hunk 6 above)
Comment (3mar-sabouni) [reply models/stock_warehouse.py]: Done boss..

**Review by mds-odoo — APPROVED** [review]

**Pull request merged**